pkg/gui/status/status_manager.go

```diff
@@ -26,39 +44,17 @@ func NewStatusManager() *StatusManager {
 	return &StatusManager{}
 }
 
-func (self *StatusManager) WithWaitingStatus(message string, f func()) {
-	self.mutex.Lock()
+func (self *StatusManager) WithWaitingStatus(message string, renderFunc func(), f func(*WaitingStatusHandle)) {
+	handle := &WaitingStatusHandle{self, message, renderFunc, -1}
```

Comment:
can we use field names here to be more explicit?

Reply:
Like this? defbd698cbc6